### ISSUE TITLE
Update `OpenAIEmbeddings` to work with OpenAI SDK updates

### ISSUE THREAD

**Issue by homanp** [langchain-ai/langchain#12970]
### System Info

LC: `v0.0.331rc1`

### Who can help?

The latest LC release candidate does not support the new embeddings API of the OpenAI SDK. 

```
AttributeError: module 'openai' has no attribute 'Embedding'
```

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Install the latest release candidate and use OpenAIEmbeddings

### Expected behavior

Update to work with latest OpenAI SDK.

**Comment by hinthornw** [comment]
Should be fixed in https://github.com/langchain-ai/langchain/pull/12969

**Comment by nubgamerz** [comment]
> Should be fixed in #12969

I have same problem as OP, using langchain == 0.0.331rc1
```
embedding_function=OpenAIEmbeddings(), persist_directory="./chroma_db_oai"
                       ^^^^^^^^^^^^^^^^^^
  File "pydantic\main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic\main.py", line 1050, in pydantic.main.validate_model
  File "G:\Code\Langchain\.venv\Lib\site-packages\langchain\embeddings\openai.py", line 284, in validate_environment
    values["client"] = openai.Embedding
                       ^^^^^^^^^^^^^^^^
AttributeError: module 'openai' has no attribute 'Embedding'. Did you mean: 'embeddings'?
```

**Comment by homanp** [comment]
> > Should be fixed in #12969
> 
> I have same problem as OP, using langchain == 0.0.331rc1
> 
> ```
> embedding_function=OpenAIEmbeddings(), persist_directory="./chroma_db_oai"
>                        ^^^^^^^^^^^^^^^^^^
>   File "pydantic\main.py", line 339, in pydantic.main.BaseModel.__init__
>   File "pydantic\main.py", line 1050, in pydantic.main.validate_model
>   File "G:\Code\Langchain\.venv\Lib\site-packages\langchain\embeddings\openai.py", line 284, in validate_environment
>     values["client"] = openai.Embedding
>                        ^^^^^^^^^^^^^^^^
> AttributeError: module 'openai' has no attribute 'Embedding'. Did you mean: 'embeddings'?
> ```

They haven't updated the rc with the fix.

**Comment by nubgamerz** [comment]
So why is the issue closed if it's not fixed in a release?

**Comment by homanp** [comment]
> So why is the issue closed if it's not fixed in a release?

Accidentally hit close

**Comment by homanp** [comment]
> Should be fixed in #12969

@hinthornw will this be released today?
